packages/components/nodes/responsibleAI/ResponsibleAI.ts

```diff
@@ -0,0 +1,30 @@
+import { BaseLanguageModel } from 'langchain/base_language'
+import { Server } from 'socket.io'
+
+export abstract class ResponsibleAI {}
+
+export abstract class Moderation extends ResponsibleAI {
+    abstract checkForViolations(llm: BaseLanguageModel, input: string): Promise<string>
+}
+
+export const checkInputs = async (inputModerations: Moderation[], llm: BaseLanguageModel, input: string): Promise<string> => {
+    for (const moderation of inputModerations) {
+        input = await moderation.checkForViolations(llm, input)
+    }
+    return input
+}
+
+// is this the correct location for this function?
+// should we have a utils files that all node components can use?
+export const streamResponse = (isStreaming: any, response: string, socketIO: Server, socketIOClientId: string) => {
```

Comment:
for now we can keep this here as its still specific to this moderation implementation. we can move this out whenever we encounter the needs of this streaming in future